src/js/firefoxandroid.js

```diff
@@ -22,11 +22,11 @@
  */
 
 require.scopes.firefoxandroid = (function() {
-var isFirefoxAndroid = !(
-  chrome.browserAction.setBadgeText &&
+var hasPopupSupport = !(
```

Comment:
Could either `hasPopupSupport` or `hasBadgeSupport` be false outside of Firefox for Android? If it could, should we add an additional guard (wherever makes the most sense) to only do the extra stuff when popup (or badge) support is missing AND we are on Firefox for Android?

Reply:
Those support checks can't be false on another browsers, at least not following the [MDN support table](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/Browser_support_for_JavaScript_APIs)

Reply:
The table for `chrome.browserAction` shows `getPopup` not supported in Edge. So we would have `hasPopupSupport` as false there, which may not be what we want.

Reply:
We don't support Edge, but it's being looked into: #1290. We should probably make the "hasSupport" checks as targeted as possible to avoid triggering Firefox on Android-specific logic elsewhere.